libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt

```diff
@@ -127,6 +129,19 @@ private fun NavigationStatus.getRouteProgress(
                             route.directionsRoute.stepGeometryToPoints(legStep)
                         }
                         routeProgressCurrentState = routeState.convertState()
+
+                        stepGeometryIndex = shapeIndex - route.directionsRoute
+                            .stepsGeometryToPoints()[legIndex]
+                            .subList(0, stepIndex)
+                            .sumOf { it.size - 1 }
```

Comment:
We should not subtract one from the number of points if `stepIndex == steps.lastIndex` because for the last step we don't want to filter out the duplicates (since there are no duplicates, there's no following adjacent step).

Reply:
Hmm, I guess this is irrelevant because the last step is never part of this sum function since `subList` is exclusive on the right side.

Reply:
+
I added `NavigatorMapperStepGeometryIndexTest`. You can look at the expected values there, I think they look valid.